### DISCUSSION TITLE
I made a package to patch `httpx` in pyodide

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [pyodide/pyodide#4999] [Show and tell]
After resolving #4954, I organized these patching code into a separated package `pyodide-httpx`.

- Repository: [CNSeniorious000/pyodide-httpx](https://github.com/CNSeniorious000/pyodide-httpx)
- PyPI: [pyodide-httpx](https://pypi.org/project/pyodide-httpx/)

Which exposes a similar API like `pyodide-http`:

```py
from pyodide_httpx import patch_httpx

patch_httpx()
```

> Note that you should also install `ssl` because [`httpx` uses it for typing internally](https://github.com/search?q=repo%3Aencode%2Fhttpx%20ssl&type=code), which I think is unnecessary
> They should use `from __future__ import annotations` for type-only imports

**Comment by CNSeniorious000** [comment]
Test code:

```py
from micropip import install
await install(["pyodide-httpx", "ssl"])

from pyodide_httpx import *
patch_httpx()

from httpx import get
get("https://httpbin.org/json")

_.json()
```

[🚀 **Run these code on py3.online**](https://py3.online/console#ZnJvbSBtaWNyb3BpcCBpbXBvcnQgaW5zdGFsbAphd2FpdCBpbnN0YWxsKFsicHlvZGlkZS1odHRweCIsICJzc2wiXSkKCmZyb20gcHlvZGlkZV9odHRweCBpbXBvcnQgKgpwYXRjaF9odHRweCgpCgpmcm9tIGh0dHB4IGltcG9ydCBnZXQKZ2V0KCJodHRwczovL2h0dHBiaW4ub3JnL2pzb24iKQoKcHJpbnQoXy50ZXh0KQpfLmpzb24oKQo=)

**Comment by ryanking13** [comment]
Awesome! Do you have plans to upstream it to `httpx` or merge it with `pyodide-http`? (cc: @koenvo).

**Reply by koenvo** [reply]
Happy to merge it into pyodide-http!

**Reply by CNSeniorious000** [reply]
Good idea. One question:

In `pyodide-httpx`, I used `pyfetch` with `run_sync` to support the sync APIs of `httpx`, which is a different approach against `pyodide-http`'s way (through `urllib3`). So we may need to change these. I'm happy to contribute but it may take some time.

I think we can discuss this in https://github.com/koenvo/pyodide-http/issues/37
